examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/decoder.py

```diff
@@ -289,44 +291,60 @@ def run(self, specs, value, cluster_name: str, typename: str, array: bool):
                 field_name = field.name
                 field_type = field.data_type.name
                 field_array = field.is_list
-                # chip-tool returns the field code as an integer but the test suite expects
-                # a field name.
-                # To not confuse the test suite, the field code is replaced by its field name
-                # equivalent and then removed.
+
+                target_name = field_name
+
                 if str(field_code) in value:
-                    value[field_name] = self.run(
-                        specs,
-                        value[str(field_code)],
-                        cluster_name,
-                        field_type,
-                        field_array
+                    # chip-tool returns the field code as an integer but the test suite expects
+                    # a field name.
+                    # To not confuse the test suite, the field code is replaced by its field name
+                    # equivalent and then removed.
+                    target_name = str(field_code)
+                else:
+                    # darwin-framework-tool returns the field name but with a different casing than
+                    # what the test suite expects.
+                    # To not confuse the test suite, the field name is replaced by its field name
+                    # equivalent from the spec and then removed.
+
+                    if field_name == 'Description' and 'descriptionString' in value:
+                        # "Description" is returned as "descriptionString" since 'description' is a reserved keyword
+                        # and can not be exposed as if in an objc struct.
+                        target_name = 'descriptionString'
+
+                    # If field_name starts with a sequence of capital letters (which can happen for
+                    # output of asLowerCamelCase if the original string started that way) lowercase all but the last one.
+                    target_name = re.sub(
+                        '^([A-Z]+)([A-Z])',
+                        lambda m: m.group(1).lower() + m.group(2),
+                        target_name
                     )
-                    del value[str(field_code)]
-
-                # darwin-framework-tool returns the field name but with a different casing than what
-                # the test suite expects.
-                # To not confuse the test suite, the field name is replaced by its field name
-                # equivalent from the spec and then removed.
-                wrong_casing_field_name = field_name[0].lower(
-                ) + field_name[1:]
-                if field_name not in value and field_name[0].upper() == field_name[0] and wrong_casing_field_name in value:
+
+                    # All field names in darwin-framework-tool starts lowercased.
+                    target_name = target_name[0].lower() + target_name[1:]
+
+                if target_name in value:
```

Comment:
I wonder whether `target_name` would be better named `non_spec_field_name` or `provided_field_name` or something to make it clear that this is the name we were handed, that corresponds to the field we are looking at....